web/src/user_status.ts

```diff
@@ -0,0 +1,138 @@
+import {z} from "zod";
+
+import * as channel from "./channel";
+import * as emoji from "./emoji";
+import type {EmojiRenderingDetails} from "./emoji";
+import {user_settings} from "./user_settings";
+
+export type UserStatus = z.infer<typeof user_status_schema>;
+export type UserStatusEmojiInfo = EmojiRenderingDetails & {
+    emoji_alt_code?: boolean;
+};
+
+type UserStatusEvent = {
+    id: number;
+    type: "user_status";
+    user_id: number;
+    away: boolean;
+    status_text: string;
+    emoji_name: string;
+    emoji_code: string | number;
+    reaction_type: string;
+};
```

Comment:
There are several optional properties for the event type. Looks like it is not documented accurately in the OpenAPI spec?
https://github.com/zulip/zulip/blob/458a0b3f3a200223b20836baef6bb7bb2e1745f2/zerver/lib/events.py#L638-L642

https://github.com/zulip/zulip/blob/458a0b3f3a200223b20836baef6bb7bb2e1745f2/zerver/lib/user_status.py#L27-L80

Reply:
Yeah it looks like the type for event was incorrect fixed it now! Thanks!